src/openxr/OpenXRApi.cpp

```diff
@@ -1512,65 +1513,61 @@ bool OpenXRApi::initialiseSwapChains() {
+	// With the GLES3 driver rendering happens into an 64bit floating point buffer with all rendering happening in linear color space.
+	// This buffer is then copied into the texture we supply here during the post process stage where tone mapping, glow, DOF, screenspace reflection and conversion to sRGB is optionally applied.
+	// Again we're expected to supply a 32bit uniform buffer
 
-	// We grab the first applicable one we find, OpenXR sorts these from best to worst choice..
+	// Note that OpenXR expects sRGB content if an sRGB buffer is used, else it expects linear, however sRGB buffers don't seem to work properly under GLES2.
+	// this is a problem especially on Quest where we have to use a 32bit buffer yet 8bits per color leads to awefull banding when linear color space is used.
 
-	keep_3d_linear = true; // This will only work correctly for GLES2 from Godot 3.4 onwards
+	// We start by defining a list of formats we'd like to use from most to least, doesn't help every platform has slightly different names for these...
+
+	std::vector<RequestedSwapchainFormat> requested_swapchain_formats;
 
-	Godot::print("OpenXR Swapchain Formats");
-	for (uint64_t i = 0; i < swapchainFormatCount && swapchainFormatToUse == 0; i++) {
-		// Godot::print("Found {0}\n", swapchainFormats[i]);
 #ifdef WIN32
-		/* disabling SRGB for now, we're rendering in linear color space...
-		if (swapchainFormats[i] == GL_SRGB8_ALPHA8 && video_driver == OS::VIDEO_DRIVER_GLES3) {
-			swapchainFormatToUse = swapchainFormats[i];
-			Godot::print("OpenXR Using SRGB swapchain.");
-			keep_3d_linear = false; // no the hardware will do conversions so we can supply sRGB values
-		}
-		*/
-		if (swapchainFormats[i] == GL_RGBA8) {
-			swapchainFormatToUse = swapchainFormats[i];
-			Godot::print("OpenXR Using RGBA swapchain.");
-		}
+	requested_swapchain_formats.push_back({ GL_SRGB8_ALPHA8, false });
+	requested_swapchain_formats.push_back({ GL_RGBA8, true });
 #elif ANDROID
-		/* disabling SRGB for now, we're rendering in linear color space...
-		if (swapchainFormats[i] == GL_SRGB8_ALPHA8 && video_driver == OS::VIDEO_DRIVER_GLES3) {
-			swapchainFormatToUse = swapchainFormats[i];
-			Godot::print("OpenXR Using SRGB swapchain.");
-			keep_3d_linear = false; // no the hardware will do conversions so we can supply sRGB values
-		}
-		*/
-		if (swapchainFormats[i] == GL_RGBA8) {
-			swapchainFormatToUse = swapchainFormats[i];
-			Godot::print("OpenXR Using RGBA swapchain.");
-		}
+	// TODO sRGB mode seems to not work on Quest so we'll leave it disabled until fixed
+	// This does mean that we'll continue to have banding issues in darker colors as 8bits
+	// does not provide enough precision for storing data in linear color space.
+	// requested_swapchain_formats.push_back({ GL_SRGB8_ALPHA8, false });
+	requested_swapchain_formats.push_back({ GL_RGBA8, true });
```

Comment:
@m4gr3d , both Malcolm and I have exhausted all options we can think off for making this work properly. While things work perfect both with SteamVR and Oculus desktop, including as I understand it over link, natively on the Quest `GL_SRGB8_ALPHA8` does not work as advertised forcing us to store data in linear color space which results in the banding issues we've been having.
Equally so, no matter what we try with the `XR_FB_COLOR_SPACE_EXTENSION_NAME`, we don't seem to be able to inform the Quest to treat the data as being in sRGB color space.

It doesn't look like we can use a `R10G10B10A2` buffer format for the swapchain as an alternative, that format would allow for storing color data in linear color space without issue.

As best we can tell this seems to be an issue with the Quest so we can't take it any further and I've commented out the line of code that would allow us to use the `GL_SRGB8_ALPHA8` texture format.

I suggest merging the PR as it is right now because it does fix all the issues for PCVR including using the plugin over airlink with the Quest 2.

We need to get a Meta engineer involved to diagnose this issue on the Quest side of life with this line of code un-commented.

Reply:
I think we should leave `GL_SRGB8_ALPHA8` enabled. I'm noticing that some of the SDK samples have similarly washed out colors so it may just be an issue with the runtime. I'm reaching internally to confirm.